### ISSUE TITLE
Scala Steward not closing old PRs 

### ISSUE THREAD

**Issue by milanvdm** [scala-steward-org/scala-steward-action#312]
Im running Scala Steward 0.13.0 with Github Actions on a private repository.

I see the following output:
```
  2021-12-21 15:26:55,030 INFO  Create branch update/helpers-clients-1.0.5379
  2021-12-21 15:26:55,158 INFO  Push 1 commit(s)
  2021-12-21 15:26:57,170 INFO  Create PR update/helpers-clients-1.0.5379
  2021-12-21 15:27:00,137 INFO  Created PR https://github.com/xxx-org/yyy/pull/551
  2021-12-21 15:27:00,151 INFO  Process update com.xxx:zzz-analysis : 0.1.7 -> 0.1.26
  ```

So it created a new PR for the helper-client but it didnt close the old PR.

From https://github.com/scala-steward-org/scala-steward/pull/1841#issuecomment-751499619, it looks like it should normally also log an INFO statement that it is checking for old PRs or branches, which it doesnt seem to do.

Am I missing something or is there actually something going wrong with Scala Steward itself?

**Comment by alejandrohdezma** [comment]
The workspace has to be removed since it is encouraged by GitHub that anything created by GitHub Actions is removed after the action has been run, so hosted runners get properly cleaned up after each run. That's why Scala Steward's workspace is stored using GH Actions own cache here:

```
2021-12-22T00:14:12.6894730Z ##[group]Saving workspace to cache...
2021-12-22T00:14:12.8644441Z [command]/usr/bin/tar --posix --use-compress-program zstd -T0 -cf cache.tzst -P -C /runner/_work/scala-steward/scala-steward --files-from manifest.txt
2021-12-22T00:14:13.1552216Z Cache Size: ~0 MB (13683 B)
2021-12-22T00:14:13.4565384Z Cache saved successfully
2021-12-22T00:14:13.4566285Z Scala Steward workspace contents saved to cache
2021-12-22T00:14:13.4567740Z ##[endgroup]
2021-12-22T00:14:13.4713035Z Post job cleanup.
```

@milanvdm can you check if you see a message like `Scala Steward workspace contents restored from cache` when running the action? Notice this step will only be done if the `cache-ttl` input is not `0`.

**Comment by milanvdm** [comment]
@alejandrohdezma Ive got the following logs:

```
2021-12-22T00:13:58.2858037Z ✓ Scala Steward workspace created
2021-12-22T00:13:58.2861896Z ##[group]Trying to restore workspace contents from cache...
2021-12-22T00:13:58.7370031Z Received 13677 of 13677 (100.0%), 0.2 MBs/sec
2021-12-22T00:13:58.7372066Z Cache Size: ~0 MB (13677 B)
2021-12-22T00:13:58.7386823Z [command]/usr/bin/tar --use-compress-program zstd -d -xf /runner/_work/_temp/51d3de55-ade2-482b-ba1c-57ff1b37705f/cache.tzst -P -C /runner/_work/scala-steward/scala-steward
2021-12-22T00:13:58.7502186Z Cache restored successfully
2021-12-22T00:13:58.7505661Z Scala Steward workspace contents restored from cache
```

Will the log line `Looking to close obsolete PRs for xxx` always show? Or only when it has a restored workspace?

**Comment by fthomas** [comment]
> Will the log line `Looking to close obsolete PRs for xxx` always show? Or only when it has a restored workspace?

It doesn't log this message anymore. [This message](https://github.com/scala-steward-org/scala-steward/blob/549b4d6e212e2a9195287768ccfadc828b976fdd/modules/core/src/main/scala/org/scalasteward/core/nurture/NurtureAlg.scala#L126) will be the first sign of Scala Steward attempting to close obsolete PRs. It will only be shown if there is a previous PR and a record of it is in the restored workspace.

**Comment by milanvdm** [comment]
Setting the `cache-ttl` to 2 days, made Scala Steward close the old PRs :)

**Comment by fthomas** [comment]
> can you check if you see a message like `Scala Steward workspace contents restored from cache` when running the action? Notice this step will only be done if the `cache-ttl` input is not `0`.

I think it should also be done if it is set to `0`. If one sets `cache-ttl` to `0` to force Scala Steward to always check for new versions, it should not imply that it can't close obsolete PRs anymore.

WDYT @alejandrohdezma ?

**Comment by milanvdm** [comment]
The action prints `Scala Steward workspace contents restored from cache` now with the cache-ttl set higher than the cron expression it runs on :)

**Comment by alejandrohdezma** [comment]
> I think it should also be done if it is set to 0. If one sets cache-ttl to 0 to force Scala Steward to always check for new versions, it should not imply that it can't close obsolete PRs anymore.
>
> WDYT @alejandrohdezma ?

Yeah, I agree. I'll update it.

**Comment by alejandrohdezma** [comment]
This change has been released on `v2.16.0`. `v2` tag has been updated too.
